internal/deb/extract.go

```diff
@@ -246,11 +246,22 @@ func extractData(dataReader io.Reader, options *ExtractOptions) error {
 				}
 			}
 			// Create the entry itself.
+			link := tarHeader.Linkname
+			if tarHeader.Typeflag == tar.TypeLink {
+				// The hard link requires the real path of the target file.
+				// For a hard link, we assume:
+				// - the target file is already created (the target file should
+				//   exist before the hard link in a valid tarball)
```

Comment:
Have a look at the comment above which describes something very similar:
```go
	// The assumption is that the tar entries of the parent directories appear
	// before the entry for the file itself. This is the case for .deb files but
        // not for all tarballs.
```

Reply:
This part is removed and new comments added at `fsutil.Create`